src/overlays/actors/ovl_Boss_Dodongo/z_boss_dodongo.c

```diff
@@ -51,311 +44,1718 @@ const ActorInit Boss_Dodongo_InitVars = {
+        dy = currentExplosive->world.pos.y - this->mouthPos.y;
+        dz = currentExplosive->world.pos.z - this->mouthPos.z;
+
+        if ((fabsf(dx) < 40.0f) && (fabsf(dy) < 40.0f) && (fabsf(dz) < 40.0f)) {
+            Actor_Kill(currentExplosive);
+            return true;
+        }
+
+        currentExplosive = currentExplosive->next;
+    }
+
+    return false;
+}
+
+void BossDodongo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BossDodongo* this = THIS;
+    s16 i;
+    u16* temp_s1_3;
+    u16* temp_s2;
+    u32 temp_v0;
+
+    globalCtx->specialEffects = &this->effects;
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 9200.0f, ActorShadow_DrawCircle, 250.0f);
+    Actor_SetScale(&this->actor, 0.01f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_0601B310, &D_0600F0D8, NULL, NULL, 0);
+    Animation_PlayLoop(&this->skelAnime, &D_0600F0D8);
+    this->unk_1F8 = 1.0f;
+    BossDodongo_SetupIntroCutscene(this, globalCtx);
+    this->health = 12;
+    this->fogMin = 995.0f;
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->fogMax = 1000.0f;
+    this->unk_224 = 2.0f;
+    this->unk_228 = 9200.0f;
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit, this->items);
+
+    if (Flags_GetClear(globalCtx, globalCtx->roomCtx.curRoom.num)) { // KD is dead
+        temp_s1_3 = SEGMENTED_TO_VIRTUAL(&D_030021D8);
+        temp_s2 = SEGMENTED_TO_VIRTUAL(gBossDodongoLavaFloorRockTex);
+
+        Actor_Kill(&this->actor);
+        Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_WARP1, -890.0f, -1523.76f,
+                           -3304.0f, 0, 0, 0, 0);
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_BREAKWALL, -890.0f, -1523.76f, -3304.0f, 0, 0, 0, 0x6000);
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_ITEM_B_HEART, -690.0f, -1523.76f, -3304.0f, 0, 0, 0, 0);
+
+        for (i = 0; i < 2048; i++) {
+            temp_v0 = i;
+            temp_s1_3[temp_v0] = temp_s2[temp_v0];
+        }
+    }
+
+    this->actor.flags &= ~1;
+}
+
+void BossDodongo_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BossDodongo* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+void BossDodongo_SetupIntroCutscene(BossDodongo* this, GlobalContext* globalCtx) {
+    s16 frames = Animation_GetLastFrame(&D_0600F0D8);
+
+    Animation_Change(&this->skelAnime, &D_0600F0D8, 1.0f, 0.0f, frames, 0, -10.0f);
+    this->actionFunc = BossDodongo_IntroCutscene;
+    this->csState = 0;
+    this->unk_1BC = 1;
+}
+
+void BossDodongo_IntroCutscene(BossDodongo* this, GlobalContext* globalCtx) {
+    f32 phi_f0;
+    Camera* camera;
+    Player* player;
+    Vec3f sp60;
+    Vec3f sp54;
+    Vec3f sp48;
+
+    player = PLAYER;
+    camera = Gameplay_GetCamera(globalCtx, 0);
```

Comment:
believe there are new defines with the camera, check the other bosses or usages of this func